Create cognitive services

BassOfLion opened the pull request:
Add module for cognitive services

Commit:
Add cognitive account

To be continue..

Commit:
Add main.tf tests and README

Commit:
Add output.tf refactoring

Commit:
Refactor and add support for map of cog services

yngveh commented on README.md:
```diff
@@ -1,2 +1,74 @@
+
+## Usage
+
+Example that shows configuration for deployment of cogtnitive service Face API 
```
cogtnitive -> cognitive

yngveh commented on examples/diagnostics/main.tf:
```diff
@@ -0,0 +1,30 @@
+
+  diagnostics = {
+    destination   = "test"
+    eventhub_name = "diagnostics",
```
Remove "," after lines.

yngveh commented on main.tf:
```diff
@@ -0,0 +1,98 @@
+  kind                = each.value.kind
+  sku_name            = each.value.sku_name
+  tags                = var.tags
+}
```
Add network_acls. Should any other attributes be added as well ?

BassOfLion replied:
custom_subdomain_name is required if network_acls is used even though it is specified as optional. Issue registered here https://github.com/hashicorp/terraform-provider-azurerm/issues/10041

Commit:
Resolve issues

Commit:
Add ip-filtering

Because of terraform provider bug custom_subdomain_name property has to be set if network_acls is used. If not there will be an error (at least in tests) https://github.com/hashicorp/terraform-provider-azurerm/issues/10041

Commit:
Convert network_acls block to optional
